### PR TITLE
Replace element name validation library

### LINKED ISSUE #64: Security issues with dependency
There are some high severity issues after running npm audit.

```
❯ npm audit                                                                                                                    (master|✔)
# npm audit report

normalize-url  4.3.0 - 4.5.0 || 5.0.0 - 5.3.0 || 6.0.0
Severity: high
Regular Expression Denial of Service - https://npmjs.com/advisories/1755
fix available via `npm audit fix`
node_modules/normalize-url

trim-newlines  <3.0.1 || =4.0.0
Severity: high
Regular Expression Denial of Service - https://npmjs.com/advisories/1753
fix available via `npm audit fix --force`
Will install validate-element-name@3.0.0, which is a breaking change
node_modules/trim-newlines
  meow  3.4.0 - 5.0.0
  Depends on vulnerable versions of trim-newlines
  node_modules/meow
    validate-element-name  2.0.0 - 2.1.1
    Depends on vulnerable versions of meow
    node_modules/validate-element-name

4 high severity vulnerabilities
```

I can try to create a PR to update some dependencies.

### PR DESCRIPTION
**Pull request by fernandopasik**
Fixes #64 

Use `is-valid-element-name` for validating element name and check for best practices

### CODE REVIEW

File: src/rules/no-invalid-element-name.ts [hunk 1]
```diff
@@ -5,18 +5,29 @@
...
 import * as ESTree from 'estree';
-import validate = require('validate-element-name');
+import isValidElementName = require('is-valid-element-name');
 
 //------------------------------------------------------------------------------
 // Rule Definition
 //------------------------------------------------------------------------------
 
+export const errorMessage =
+  "Element name is invalid and should follow the HTML standard's recommendations (https://html.spec.whatwg.org/multipage/custom-elements.html#prod-potentialcustomelementname).";
```
Comment (43081j) [inline src/rules/no-invalid-element-name.ts:15]: just for code readability can you split this over two lines?

```ts
export const errorMessage = "Element name is invalid and should follow the HTML standard's recommendations" +
  " (URL_HERE)";
```

File: src/rules/no-invalid-element-name.ts [hunk 2]
```diff
@@ -60,15 +71,15 @@ const rule: Rule.RuleModule = {
...
             const isWarning =
               !(options && options.loose) &&
-              validationResult.message !== undefined;
+              !isBestPracticeElementName(firstArg.value);
 
-            if (!validationResult.isValid || isWarning) {
+            if (!validationResult || isWarning) {
               context.report({
-                message: validationResult.message,
+                message: errorMessage,
```
Comment (43081j) [inline src/rules/no-invalid-element-name.ts:82]: to make it a bit nicer to test, you can add the error message to the `meta`, like:

```ts
meta: {
  messages: {
    invalidName: "error goes here"
  }
}
```

then usage would be `messageId: 'invalidName'`

you don't need to import the error in the tests then and can just set `messageId` in the test expectation

File: src/rules/no-invalid-element-name.ts (hunk 1 above)
Comment (fernandopasik) [reply src/rules/no-invalid-element-name.ts]: Cool, I've reformatted this

File: src/rules/no-invalid-element-name.ts (hunk 2 above)
Comment (fernandopasik) [reply src/rules/no-invalid-element-name.ts]: Got it, I've added and referenced it from the meta

File: src/test/rules/no-invalid-element-name_test.ts [hunk 3]
```diff
@@ -78,8 +78,7 @@ ruleTester.run('no-invalid-element-name', rule, {
       customElements.define('my-App', MyApp)`,
       errors: [
         {
-          message:
-            'Custom element names must not contain uppercase ASCII characters.',
+          message: errorMessage,
```
Comment (43081j) [inline src/test/rules/no-invalid-element-name_test.ts:81]: these can now be `messageId: 'invalidElementName'`, then pr looks good i think

Comment (fernandopasik) [reply src/test/rules/no-invalid-element-name_test.ts]: Cool, I've replaced the `message` for `messageId` in tests and also stop exporting it from the src file

**Comment by 43081j** [comment]
@fernandopasik thanks so much for sorting this out. always good to get rid of a bloated dependency, and bonus actually fixed what we originally wanted to fix: the audits

**Review by 43081j — APPROVED** [review]

**Pull request merged**